x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -381,6 +415,30 @@ export class AlertsClient {
     }
   }
 
+  /**
+   * When an update by ids is requested, do a multi-get, ensure authz and audit alerts, then execute bulk update
+   * @param param0
+   * @returns
+   */
+  private async mgetAlertsAuditOperateStatus({
+    ids,
+    status,
+    indexName,
+    operation,
+  }: {
+    ids: string[];
+    status: STATUS_VALUES;
+    indexName: string;
+    operation: ReadOperations.Find | ReadOperations.Get | WriteOperations.Update;
+  }) {
+    return this.mgetAlertsAuditOperate({
+      ids,
+      indexName,
+      operation,
+      fieldToUpdate: (source) => this.getAlertStatusFieldUpdate(source, status),
```

Comment:
Same logic as before. I moved it out from `mgetAlertsAuditOperate`. The consumer of `mgetAlertsAuditOperate` should define how to get the field based on the source.